### DISCUSSION TITLE
How do I give my module a pre-release version number

### DISCUSSION THREAD

**Discussion by stephanjohnson** [IntentArchitect/Support#41] [Q&A]
I have tried setting the version number to include a text component after the patch component of the version number, however when I do this the SF does not detect the change at all and shows a successful build. This build does not reflect as a module file in my output location, and as a result, does not reflect in the Module Manager.

How do I set a version number to include a text component as the standard Intent modules do by adding an `-pre.n` or `-alpha.n` postfix? It also does not seem like there is an option to set the entire module as prerelease.

**Comment by joelsteventurner** [comment]
Hi @stephanjohnson 

As you may be aware, Modules use [Semantic Versioning 2.0.0](https://semver.org/spec/v2.0.0.html), meaning a module is regarded as a pre-release solely by its version having a hyphen in it.

Making the version a pre-release in the Module Builder is good enough, but the likely reason for it not seeming to have an effect is that the `.imodspec` was already generated and the Module Builder does not overwrite the version in the `.imodspec` with a lower version from the Module Builder. This means that if the `.imodspec` has `1.0.0` and in the Module Builder version `1.0.0-pre.0` is specified, as it's lower than the `.imodspec` version, it will not overwrite it. In such cases you can manually change the version in the `.imodspec` file and then going forward it should work from the Module Builder so long as you're not trying to output a lower version again.

**Comment by stephanjohnson** [comment]
Hi @joelsteventurner,

That worked. I ended up going into the `.imodspec` file and the Module Builder designer and reset the version number to a lower version number, and after that everything is working as expected now.
